File: 39.js
function countNiceSubarrays(nums, k) {
    let left = 0, sum = 0, ans = 0;

    for (let right = 0; right < nums.length; right++) {
        sum += nums[right];

        while (left <= right && sum >= k) {
            sum -= nums[left];
            left++;
        }

        ans += (right - left + 1);
    }
    return ans;
}
// Driver code
const fs = require("fs");
const input = fs.readFileSync(0, "utf8").trim().split("\n");
const nums = input[0].split(" ").map(Number);
const k = parseInt(input[1], 10);
console.log(countNiceSubarrays(nums, k));

File: 39.py
def countNiceSubarrays(nums, k):
    left = 0
    curr = 0
    ans = 0

    for right in range(len(nums)):
        curr += nums[right]

        while left <= right and curr >= k:
            curr -= nums[left]
            left += 1

        ans += (right - left + 1)

    return ans
# Example usage:
nums = [1, 2, 3, 4, 5,6,7]
k = 5
print(countNiceSubarrays(nums, k))  # Output: 9
